pi/heartbeat.py

```diff
@@ -192,6 +194,10 @@ def send_heartbeat(flic_last_seen_secs, flic_last_ping_secs, system_id):
                 num_secs_darkstat = parse_flic_last_seen_from_darkstat_html(html, FLIC_MAC_ADDRESS)
                 num_secs_ping = (datetime.datetime.now() - last_ping).total_seconds()
                 system_ok = send_heartbeat(num_secs_darkstat, num_secs_ping, system_id)
```

Comment:
Can this call take a long time? Would you want to re-run all of
```
                html = get_darkstat_html()
                ip = parse_flic_ip_from_darkstat_html(html, FLIC_MAC_ADDRESS)
                if ping(ip):
                    last_ping = datetime.datetime.now()
                num_secs_darkstat = parse_flic_last_seen_from_darkstat_html(html, FLIC_MAC_ADDRESS)
                num_secs_ping = (datetime.datetime.now() - last_ping).total_seconds()
```
inside the `if not system_ok:` section as well?

Reply:
hm, I see what you mean. maybe I can do a small refactor of that block into another function and run it again inside the retry block.